src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs

```diff
@@ -205,6 +268,54 @@ internal static void RegisterOtlpExporterOptionsFactory(IServiceCollection servi
                 sp.GetRequiredService<IOptionsMonitor<BatchExportActivityProcessorOptions>>().Get(name)));
     }
 
+    internal HttpClientHandler CreateDefaultHttpMessageHandler()
+    {
+        var handler = new HttpClientHandler();
+
+        if (this.CertificateFile != null)
+        {
+#if NET6_0_OR_GREATER
+            var trustedCA = LoadCertificateFromPemFile(this.CertificateFile, null);
+
+            handler.ServerCertificateCustomValidationCallback = (message, serverCert, chain, policyErrs) =>
+            {
+                chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
+                chain.ChainPolicy.CustomTrustStore.Add(trustedCA);
+                var buildResult = chain.Build(serverCert);
+                return buildResult;
+            };
```

Comment:
I'm not an expert on these things, so still need to do some of my own research to understand if this is sufficient.

One thing that leaves me a bit unsure here is that `policyErrs` is ignored. My understanding is the "out-of-the-box" validations would be reflected in `policyErrs` (e.g., is the cert signed by a trusted CA, is the cert expired, etc). I suspect we want the custom validation here to be in addition to the out-of-the-box validations.

Reply:
You're correct that `policyErrs` would contain the error that the "default" validation method. The default validator would by default trust the certificates in the root trust store (aka certificates "installed" on the machine).  
If user is manually trusting a certificate, it means the certificate might not be in the machine root store and we need to at least ignore that error.

The other validation criteria (e.g. expiration, revocation) is encoded in the [`X509Chain` parameter](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509chain?view=net-7.0). In this implementation I keep them as default setting. This means that other failure criteria would still work (e.g. the trusted cert still fails if it's expired).